src/Core/Settings.cpp

```diff
@@ -5916,7 +5916,6 @@ Experimental data deduplication for SELECT queries based on part UUIDs
     MAKE_OBSOLETE(M, UInt64, parallel_replicas_min_number_of_granules_to_enable, 0) \
     MAKE_OBSOLETE(M, ParallelReplicasCustomKeyFilterType, parallel_replicas_custom_key_filter_type, ParallelReplicasCustomKeyFilterType::DEFAULT) \
     MAKE_OBSOLETE(M, Bool, query_plan_optimize_projection, true) \
-    MAKE_OBSOLETE(M, Bool, query_cache_store_results_of_queries_with_nondeterministic_functions, false) \
```

Comment:
Why do we want to remove it? Anybody still using this setting in their queries will get an error instead of it simply being ignored.

Reply:
Yes, that comment itself it's obsolete. It exists from a time when compatibility was broken constantly and compatibility changes did not exists. Let's remove the comment instead.

Once we stop using deprecated settings in code (https://github.com/ClickHouse/ClickHouse/pull/71148) I can make it inexpensive to keep them around.